plugins/terraform/layer0/resource_layer0_environment_test.go

```diff
@@ -0,0 +1,100 @@
+package layer0
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/api/job"
+	"github.com/quintilesims/layer0/client/mock_client"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResourceEnvironmentCreateRead(t *testing.T) {
```

Comment:
We should add some test cases for all the entities to capture the https://www.terraform.io/docs/plugins/provider.html#exists behavior Josh mentioned elsewhere -- i.e. someone deletes an entity from Swagger or the CLI, and terraform can gracefully handle it when running `apply`, etc.